test/e2e/tests/vault-corruption/vault-corruption.spec.ts

```diff
@@ -0,0 +1,368 @@
+
+    return firstAddress;
+  }
+
+  /**
+   * Returns the vault recovery button.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function getVaultRecoveryButton(driver: Driver) {
+    return await driver.waitForSelector('#critical-error-button', {
+      // The button is disabled for a timeout period as a precaution, so we
+      // need to wait for it to be enabled before continuing.
+      state: 'enabled',
+    });
+  }
+
+  /**
+   * Click the recovery/reset button then confirm or dismiss the action.
+   *
+   * @param driver - The WebDriver instance.
+   * @param confirm - Whether to confirm the action or not.
+   */
+  async function clickRecover(driver: Driver, confirm: boolean) {
+    // click the Recovery/Reset button
+    const recoveryButton = await getVaultRecoveryButton(driver);
+    await recoveryButton.click();
+
+    // Confirm we want to recover/reset.
+    const prompt = await driver.driver.switchTo().alert();
+    if (confirm) {
+      await prompt.accept();
+    } else {
+      await prompt.dismiss();
+    }
+
+    // the button should be disabled while the recovery process is in progress,
+    // and enabled if the user dismissed the prompt
+    const isEnabled = await recoveryButton.isEnabled();
+    assert.equal(
+      isEnabled,
+      !confirm,
+      confirm
+        ? 'Recovery button should be disabled when prompt is accepted'
+        : 'Recovery button should be enabled when prompt is dismissed',
+    );
+  }
+
+  /**
+   * Recovers the vault and returns the first account's address.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function onboardAfterRecovery(driver: Driver) {
+    // Log back in to the wallet
+    const loginPage = new LoginPage(driver);
+    await loginPage.check_pageIsLoaded();
+    await loginPage.loginToHomepage(WALLET_PASSWORD);
+
+    // complete metrics onboarding flow
+    await onboardingMetricsFlow(driver, {
+      participateInMetaMetrics: false,
+      dataCollectionForMarketing: false,
+    });
+
+    const secureWalletPage = new SecureWalletPage(driver);
+    await secureWalletPage.check_pageIsLoaded();
+    await secureWalletPage.skipSRPBackup();
+
+    // finish up onboarding screens
+    const onboardingCompletePage = new OnboardingCompletePage(driver);
+    await onboardingCompletePage.check_pageIsLoaded();
+    await onboardingCompletePage.completeOnboarding();
+
+    // it should take us to the home page automatically
+    const homePage = new HomePage(driver);
+    homePage.check_pageIsLoaded();
+
+    // since our state has been reset, but we skip the welcome screen we now
+    // need to accept the terms of use again
+    const updateTermsOfUseModal = new TermsOfUseUpdateModal(driver);
+    await updateTermsOfUseModal.check_pageIsLoaded();
+    await updateTermsOfUseModal.confirmAcceptTermsOfUseUpdate();
+
+    // finally, get the first account's address
+    return await getFirstAddress(driver);
+  }
+
```

Comment:
a small suggestion, but non blocking: we could move the above functions into [a flow file](https://github.com/MetaMask/metamask-extension/tree/main/test/e2e/page-objects/flows). Maybe using the existing `onboarding.flow.ts` file, seems a good fit :thinking: what do you think?
This way we could re-use them in other tests if needed -- though it does not seem very likely now, so up to you :)

Reply:
How's this? https://github.com/MetaMask/metamask-extension/pull/32006/files#diff-ec7bab9cc504858929bf992671dbdeb049a40e8feda15dce532b313a6e89df5eR355-R398

Reply:
great, thank you!!